### DISCUSSION TITLE
How to see console capsule output when the process_console capsule is running

### DISCUSSION THREAD

**Discussion by potto216** [tock/tock#3348] [Q&A] [answered]
How do you see console capsule output when restarting an application with the process_console capsule? I am using an nRF52840 board with the example console app in libtock-rs. When I run `make flash-nrf52840 EXAMPLE=console` the app is built (I made a minor mod to the output string) and installed on the board and I can select the Segger com port to watch. I then get
```
Initialization complete. Entering main loop
NRF52 HW INFO: Variant: AAD0, Part: N52840, Package: QI, Ram: K256, Flash: K1024
Hello world This is the Console App!
tock$ 
```
which is what I expect. The app runs then I see the process console. however, whenever I try to rerun the app with
```
tock$ stop console
Process console stopped
tock$ start console
Process console resumed.
```
I don't see the console output of `Hello world This is the Console App!`. Only if I reboot the board do the see the console app output before the process_console runs.  

How can I restart the app in the process console and see the apps output using the Console capsule? Thanks!

**Comment by hudson-ayers** [comment, marked as answer]
Stopping and starting an app does not restart it. If your app printed something in a loop, you would see that stopping stops it, and starting it resumes it from where it left off.

I don't think there is currently a `restart` command on the process console, though that would probably be good to add. You can, in the meantime, configure the process fault policy for your board to have faulted apps restart (https://github.com/tock/tock/blob/053b90b48305609d72c4969317604179f81bb45d/boards/hail/src/main.rs#L487 is an example) and then use the fault command to force a restart

**Reply by potto216** [reply]
Thanks! I verified your suggestions by adding
```
use libtock::alarm::{Alarm, Milliseconds};

    loop {
        writeln!(Console::writer(), "Hello world This is the Console App!").unwrap();
        Alarm::sleep_for(Milliseconds(1000)).unwrap();
    }
```
to the console app and it output text every second and stopped when I issued the command 'stop console' and started back up when I issued 'start console'

I also modified `/home/user/tockos/tock/boards/nordic/nrf52840dk/src/main.rs` similar to your example and I stopped the app with `stop console` and when I issued the command 'fault console' it restarted.

**Comment by bradjc** [comment]
We also now have [terminate and boot](https://github.com/tock/tock/blob/master/doc/Process_Console.md#terminate-and-boot) in the process console now.
